dashboard/src/components/DeploymentForm/DeploymentForm.tsx

```diff
@@ -221,6 +221,11 @@ class DeploymentForm extends React.Component<IDeploymentFormProps, IDeploymentFo
 
   public handleValuesChange = (value: string) => {
     this.setState({ appValues: value, valuesModified: true });
+    if (this.shouldRenderBasicForm()) {
+      this.setState({
+        basicFormParameters: retrieveBasicFormParams(value, this.props.selected.schema),
+      });
+    }
```

Comment:
Looks like this will mean the basic form params are updated on every keystroke into the text box? Might be fine as is, I was just wondering if it was necessary (ie. could we just do this once when switching to the basic form).

Also, I've not checked, but assuming this doesn't create a loop (ie. if I update the basic form, it updates the values which triggers this method, which re-evaluates the basic parameters). From memory you're using the string value as the source of truth, so it may do (not quite a loop, but perhaps not wanted).

Reply:
> Looks like this will mean the basic form params are updated on every keystroke into the text box? Might be fine as is, I was just wondering if it was necessary (ie. could we just do this once when switching to the basic form).

Well, yes, it may be an issue if the values is big, I can move this logic whenever the `Basic` tab is clicked.

> Also, I've not checked, but assuming this doesn't create a loop (ie. if I update the basic form, it updates the values which triggers this method, which re-evaluates the basic parameters). From memory you're using the string value as the source of truth, so it may do (not quite a loop, but perhaps not wanted).

Good point, there was indeed an unnecessary step.